src/app/pages/config-page/config-page.component.ts

```diff
@@ -217,18 +222,61 @@ export class ConfigPageComponent implements OnInit, OnDestroy {
     this._subs.unsubscribe();
   }
 
-  saveGlobalCfg($event: {
+  async saveGlobalCfg($event: {
     sectionKey: GlobalConfigSectionKey | ProjectCfgFormKey;
     config: any;
-  }): void {
+  }): Promise<void> {
     const config = $event.config;
     const sectionKey = $event.sectionKey as GlobalConfigSectionKey;
 
     if (!sectionKey || !config) {
       throw new Error('Not enough data');
-    } else {
-      this.configService.updateSection(sectionKey, config);
     }
+
+    // Check if user is trying to disable user profiles when multiple profiles exist
+    if (
+      sectionKey === 'misc' &&
+      config.isEnableUserProfiles === false &&
+      this._userProfileService.hasMultipleProfiles()
+    ) {
+      const currentMiscConfig = this.globalCfg?.misc;
+      // Only show dialog if we're actually changing from true to false
+      if (currentMiscConfig?.isEnableUserProfiles !== false) {
```

Comment:
I think this would be more reliable, because `currentMiscConfig` can be `undefined`

```suggestion
      if (currentMiscConfig?.isEnableUserProfiles === true) {
```